rate-providers/sUSDSBaseRateProvider.md

```diff
@@ -0,0 +1,49 @@
+# Rate Provider: `SavingsUSDSRateProvider`
+
+## Details
+- Reviewed by: @mkflow27
+- Checked by: @\<GitHub handle of secondary reviewer\>
+- Deployed at:
+    - [base:0x84394fa6a39bdff63b255622da362b113c690267](https://basescan.org/address/0x84394fa6a39bdff63b255622da362b113c690267#code)
+- Audit report(s):
+    - [Chainsecurity audit](https://docs.spark.fi/assets/Chainsecurity-sUSDS.pdf)
+
+## Context
+This rate Provider is providing a bridged rate from the L1 sUSDS contract. It briges the rate (`chi`) via the Base message bridge. 
+
+## Review Checklist: Bare Minimum Compatibility
+Each of the items below represents an absolute requirement for the Rate Provider. If any of these is unchecked, the Rate Provider is unfit to use.
+
+- [x] Implements the [`IRateProvider`](https://github.com/balancer/balancer-v2-monorepo/blob/bc3b3fee6e13e01d2efe610ed8118fdb74dfc1f2/pkg/interfaces/contracts/pool-utils/IRateProvider.sol) interface.
+- [x] `getRate` returns an 18-decimal fixed point number (i.e., 1 == 1e18) regardless of underlying token decimals.
+
+## Review Checklist: Common Findings
+Each of the items below represents a common red flag found in Rate Provider contracts.
+
+If none of these is checked, then this might be a pretty great Rate Provider! If any of these is checked, we must thoroughly elaborate on the conditions that lead to the potential issue. Decision points are not binary; a Rate Provider can be safe despite these boxes being checked. A check simply indicates that thorough vetting is required in a specific area, and this vetting should be used to inform a holistic analysis of the Rate Provider.
+
+### Administrative Privileges
+- [ ] The Rate Provider is upgradeable (e.g., via a proxy architecture or an `onlyOwner` function that updates the price source address). 
+
+- [x] Some other portion of the price pipeline is upgradeable (e.g., the token itself, an oracle, or some piece of a larger system that tracks the price).
+    - upgradeable component: `SUsds` ([ethereum:0xa3931d71877C0E7a3148CB7Eb4463524FEc27fbD](https://etherscan.io/address/0xa3931d71877C0E7a3148CB7Eb4463524FEc27fbD#readProxyContract))
```

Comment:
How did you get to these L1 addresses?

Reply:
You can see that an `onlyRole(DATA_PROVIDER_ROLE)` can call a function that does
```
function _setSUSDSData(ISSROracle.SUSDSData memory nextData) internal {
    _data = nextData;

    emit SetSUSDSData(nextData);
}
```
The upgrader is deployed at: https://basescan.org/address/0x212871A1C235892F86cAB30E937e18c94AEd8474#code and has an `l1Authority`. You can investigate the `refresh` method there and see that it reads from the susds contract and bridges it.

Reply:
Wow - it was quite an effort to follow all these steps 😂 
1. Identify you needed DATA_PROVIDER_ROLE to be able to update the data
2. Check transactions to find a `grantRole` to `0x212871A1C235892F86cAB30E937e18c94AEd8474`
3. See that it has an l1Authority (that shows up as EOA on Base)
4. Check for that same address on L1 to see it has a contract implementation
5. See that its implementation has an `l2Oracle` that points to the data provider of the L2
I think I got the whole flow 👍 
Thanks for explaining 🙏

Reply:
Ps: at first I didn't know the `keccak` transformation of `DATA_PROVIDER_ROLE` was available within the `Read Contract` tab, but I just learned that it's trivial to generate that using `cast keccak DATA_PROVIDER_ROLE` from foundry